apps/withdraw-pool/src/components/PoolForm.tsx

```diff
@@ -0,0 +1,30 @@
+import { useUserPoolBalance } from "#/hooks/useUserPoolBalance";
+import { Spinner, useIFrameContext } from "@bleu/cow-hooks-ui";
+import { WithdrawPctSlider } from "./WithdrawPctSlider";
+import { PoolBalancesPreview } from "./PoolBalancePreview";
+import { SubmitButton } from "./SubmitButton";
+
+export function PoolForm({ poolId }: { poolId?: string }) {
+  const { context } = useIFrameContext();
+  const {
+    data: poolBalances,
+    isValidating,
+    isLoading,
+  } = useUserPoolBalance({
+    user: context?.account,
+    chainId: context?.chainId,
+    poolId,
+  });
+
+  if (!context || !poolId || !poolBalances || !poolBalances.length) return null;
+
+  if (isValidating || isLoading) return <Spinner size="xl" />;
```

Comment:
this loading will never render the first time since on the first load there's no `poolBalances` and thus the if clause above will return right?